src/commands/CreateEnvironmentVariableCommand.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as constants from '../constants';
+import * as extension from '../extension';
+import * as utils from '../utils';
+import * as vscode from 'vscode';
+import { DataVirtConfig, ValueFrom, ConfigMapRef, SecretRef, Property, KeyRef } from '../model/DataVirtModel';
+import { EnvironmentTreeNode } from '../model/tree/EnvironmentNode';
+
+export async function createEnvironmentVariableCommanForValue(envNode: EnvironmentTreeNode) {
```

Comment:
should this file be named slightly different since it's more than just Create? perhaps EnvironmentVariableCommands.ts?

Reply:
it is just creating environment variables. just that it can be 3 types of them. Secret, ConfigMap and Value type, but its all about creation.